zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/FlowNodeInstanceFromProcessInstanceHandler.java

```diff
@@ -90,9 +93,23 @@ public void updateEntity(
     entity.setScopeKey(recordValue.getFlowScopeKey());
 
     if (intent.equals(ELEMENT_ACTIVATING)) {
-      // we set the default value here, which may be updated later within incident export
-      entity.setTreePath(recordValue.getProcessInstanceKey() + "/" + record.getKey());
-      entity.setLevel(1);
+      final ProcessInstanceRecordValue value = record.getValue();
+      if (value.getElementInstancePath() != null && !value.getElementInstancePath().isEmpty()) {
+        // Build the treePath in the format
+        // <processInstanceKey>/<flowNodeInstanceKey>/.../<flowNodeInstanceKey> where upper level
+        // flowNodeInstanceKeys are normally subprocess(es) or multi-instance body This is an intra
+        // tree path that shows position of flow node instance inside one process instance.
+        final List<String> treePathEntries =
+            value.getElementInstancePath().getFirst().stream().map(String::valueOf).toList();
```

Comment:
❌ This might be not correct. I think it should be `getLast`, see https://github.com/camunda/camunda/commit/a05003cd1ba59ead7a376ca88771017ffdc0c2b5#diff-a07129f0655c6899a0d04214069f7a23d3a2f09b7573b6d8317099e0422d565cR1103-R1105

Reply:
Caught this in test and fixed in the next commit!